src/transport/AdminPairingTable.cpp

```diff
@@ -232,6 +243,44 @@ CHIP_ERROR AdminPairingInfo::SetRootCert(const ByteSpan & cert)
     return CHIP_NO_ERROR;
 }
 
+void AdminPairingInfo::ReleaseICACert()
+{
+    if (mICACert != nullptr)
+    {
+        chip::Platform::MemoryFree(mICACert);
+    }
+    mICACertAllocatedLen = 0;
+    mICACertLen          = 0;
+    mICACert             = nullptr;
+}
+
+CHIP_ERROR AdminPairingInfo::SetICACert(const ByteSpan & cert)
+{
+    if (cert.size() == 0)
+    {
+        ReleaseICACert();
+        return CHIP_NO_ERROR;
+    }
+
+    VerifyOrReturnError(cert.size() <= kMaxChipCertSize, CHIP_ERROR_INVALID_ARGUMENT);
+    if (mICACertLen != 0 && mICACertAllocatedLen < cert.size())
+    {
+        ReleaseICACert();
+    }
+
+    if (mICACert == nullptr)
+    {
+        mICACert = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(cert.size()));
+    }
+    VerifyOrReturnError(mICACert != nullptr, CHIP_ERROR_NO_MEMORY);
+    VerifyOrReturnError(CanCastTo<uint16_t>(cert.size()), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This `VerifyOrReturnError(CanCastTo<uint16_t>(cert.size()), CHIP_ERROR_INVALID_ARGUMENT);` needs to be moved *before* the alloc.

Reply:
Will update.